### ISSUE TITLE
ActiveSupport::TimeZone.all throws exception

### ISSUE THREAD

**Issue by kalinchuk** [rails/rails#33128]
### Steps to reproduce

`ActiveSupport::TimeZone.send(:zones_map).values` does not include the `Srednekolymsk` time zone and instead returns `nil` for it. Because of that this happens:

```
ActiveSupport::TimeZone.all
ArgumentError: comparison of NilClass with ActiveSupport::TimeZone failed
```

### Expected behavior

`ActiveSupport::TimeZone.all` should return a list of TZInfo::DateTimezone objects and not error out.

### Actual behavior

An exception is thrown during the `sort` in the `all` method due to `nil` value.

### System configuration
**Rails version**: Rails 5.2.0

**Ruby version**: ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-linux]

**Comment by y-yagi** [comment]
> ActiveSupport::TimeZone.send(:zones_map).values does not include the Srednekolymsk time zone 

Hmmm. It seems Srednekolymsk includes in zones map. 

```
$ ./bin/rails r 'p ActiveSupport::TimeZone.send(:zones_map).values.find {|v| v.name == "Srednekolymsk"}'
#<ActiveSupport::TimeZone:0x000055fa4ea9f8c8 @name="Srednekolymsk", @utc_offset=nil, @tzinfo=#<TZInfo::DataTimezone: Asia/Srednekolymsk>> 
```
Could you please explain the details of the issue?
Or can you provide an application that reproduces the issue?

**Comment by y-yagi** [comment]
I guess this issue has been fixed with #32613. 
Therefore, I will close this. However, if the issue reappears in the master branch, please reopen it freely. Thanks.
